IntegrationTests/Tab/AddressBarTests.swift

```diff
@@ -190,6 +195,53 @@ class AddressBarTests: XCTestCase {
 
     }
 
+    @MainActor
+    func testWhenAddressIsTyped_LoadedTopHitSuggestionIsCorrectlyAppendedAndSelected() {
```

Comment:
@mallexxx this test fails sometimes so there are more chances it will fail randomly on CI. To replicate the failure right-click on the diamond on the left side of the test signature and run it repeatidly (e.g. 10 times)

<img width="496" alt="Screenshot 2024-10-10 at 12 01 54 PM" src="https://github.com/user-attachments/assets/15e04b04-8e4e-44ca-a504-a57adc1ec327">